test_util/convertECL.cpp

```diff
@@ -118,40 +119,133 @@ static void printHelp() {
               << "\nIn addition, the program takes these options (which must be given before the arguments):\n\n"
               << "-h Print help and exit.\n"
               << "-l List report step numbers in the selected restart file.\n"
+              << "-g Convert file to grdecl format.\n"
+              << "-o Specify output file name (only valid with grdecl option).\n"
               << "-i Enforce IX standard on output file.\n"
               << "-r Extract and convert a specific report time step number from a unified restart file. \n\n";
 }
 
+template <typename T>
+void writeGrdeclData(std::ofstream& ofileH, const std::string& name, const std::vector<T>& array)
+{
+    ofileH << "\n" << name << std::endl;
+
+    int ncol;
+    std::string fmt_str;
+
+    int64_t data_size = static_cast<int64_t>(array.size());
+
+    if constexpr (std::is_same<T, float>::value){
+        ncol = 4;
+        fmt_str = "%10.7E";
+    } else if constexpr (std::is_same<T, int>::value){
+        ncol = 8;
+        fmt_str = "%4d";
+    } else {
+        std::cout << "unknown ???" << std::endl;
+        exit(1);
+    }
+
+    char buffer [15];
+
+    for (int64_t n = 0; n < data_size; n++){
+        std::sprintf (buffer, fmt_str.c_str(), array[n]);
+        std::string tmpstr(buffer);
+        ofileH << " " << tmpstr;
+
+        if ((n+1) % ncol == 0)
+            ofileH << "\n";
+    }
+
+    if (data_size  % ncol == 0)
+        ofileH << " /\n";
+    else
+        ofileH << " /\n";
+}
+
+void open_grdecl_output(const std::string& output_fname, const std::string& input_file, std::ofstream& ofileH)
+{
+    if (output_fname.empty()) {
+        std::filesystem::path inputfile(input_file);
+        std::filesystem::path rootName = inputfile.stem();
+        std::filesystem::path path = inputfile.parent_path();
+        std::filesystem::path grdeclfile;
+
+        grdeclfile = path / rootName += ".grdecl";
+
+        if (std::filesystem::exists(grdeclfile)) {
+            std::cout << "\nError, cant make grdecl file " << grdeclfile.string() << ". File exists \n";
+            exit(1);
+        }
+
+        ofileH.open(grdeclfile.string(), std::ios::out);
```

Comment:
Note that since we use C++17 you don't really need `.string()` here.  You can just pass the `std::filesystem::path` object directly,
```C++
ofileH.open(grdeclfile, std::ios::out);
```